language/expressions.md

```diff
@@ -695,15 +695,21 @@ The following table shows the result of searching for a LHS of a particular type
 Assignment Operator
 ---
 
-The assignment operator assigns the result of the RHS to the L-Value produced by the LHS. An L-value is a name referring to a "slot" in the current scope (that can be referenced (typically by a 
-variable) to obtain the value).
+The assignment operator assigns the result of the RHS to one or more L-values produced by the LHS expression.
+An L-value is a name referring to a named "slot" in the current scope (that is, typically a variable).
 
 * A `$` variable produces an L-value name
-* Only a Simple Name is accepted
+* Only a Simple Name is accepted, it is not allowed to assign to something in another namepace
 * Numerical L-values are not allowed (numerical variables are read-only and set by side effect
   of matching with a regular expression).
 * Assignment is an R-value
 * The value of an assignment is the value of the RHS
+* An array of L-values forms a multi-assignment where several variables can be assigned at once from the given RHS source
+* In a multi-assignment the result depends on the type of the RHS:
+  * *Array* - each variable is assigned from the RHS array's index 0-n. It is an error if there are too few values.
+  * *Hash* - each variable is assigned the value from the corresponding key in the RHS hash. It is in an error if the key is not present.
+  * *Type[Class]* - each variable is assigned the value of the corresponding variable/parameter in the referenced class. The
+    class must have been included, and the referenced variable must exist, but may have an `undef` value assigned.
```

Comment:
yes, `declared` is horrible, I don't know what the right term is, I also use `included` for this purpose when I talk about it, but some people get confused.  `added` might work yes.